airflow/api_fastapi/common/parameters.py

```diff
@@ -163,6 +164,19 @@ def __init__(
         self.model = model
         self.to_replace = to_replace
 
+    def __or__(self, other):
```

Comment:
I see why we do that.

The only thing is that If someone explicitly provide a filtering by “primary_key” this will end up falling back on the other, while it is not what the user want.


Maybe we can go around it by using 1 SortParam with all values authorised. And put that custom logic in the transform_callback. Set the transform in the body of the view. (param.transform = lambda dag: dag….).

And then use it.


Just an idea though.

i’ll do an in depth review this weekend but overall, it looks good.

Reply:
Yes, I am thinking the same when the primary key passes this will always fall to `other`. Maybe we can have an alternative approach similar to Search and Filter, creating the SortParam with the factory in this case. Do you mean something similar to this? Or just defining the `depend` method with a transform callback that the transform will be overridden in the view? 

I am not sure if this is a needed feature for multiple use cases. This was the only logical comparison I ended up with without expanding SortParam initialisation. I am okay with both moving logic to view without expanding the SortParam or expanding the SortParam to conform to the primary key case.

Reply:
maybe putting the logic into the view at this point is simpler at this point. The general case (as we have in this PR) would need more effort to cover all cases (PK) and I don’t think it’s worth at this point, that’s the only filter with this requrement.

Reply:
Moved the `__or__` logic from `SortParam` to the view in a simple way. Thanks for the quick discussion!